src/Utilities/MakeVector.hpp

```diff
@@ -0,0 +1,62 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <vector>
+
+#include "Utilities/TMPL.hpp"
+
+namespace MakeVectorImpl {
+/// \cond NEVER
+template <class T>
+struct tag {
+  using type = T;
+};
+template <class T, size_t n>
+struct get_tag : get_tag<typename T::type, n - 1> {};
+template <class T>
+struct get_tag<T, 0> : tag<T> {};
+
+template <class Tag, size_t n = 1>
+using type_t = typename get_tag<Tag, n>::type;
+/// \endcond
+
+// build the return type:
+template <class T0, class... Ts>
+using vector_types =
+    type_t<std::conditional<std::is_same<T0, void>::value,
+                            tmpl::front<tmpl::list<typename std::decay<Ts>...>>,
+                            tag<T0>>,
+           2>;
+template <class T0, class... Ts>
+using vector_return_type = std::vector<vector_types<T0, Ts...>>;
+}  // namespace MakeVectorImpl
+
+/*!
+ * \brief Constructs a `std::vector` containing each of the arguments in 'ts'.
+ *
+ * This is useful as it allows in place construction of a vector of non-copyable
+ * objects. If the types contained in 'ts' are copyable, this is identical to
+ * construction of the std::vector using an initializer list.
+ *
+ * \example
+ * \snippet Utilities/Test_MakeVector.cpp make_vector_example
+ */
+template <class T0 = void, class... Ts,
+          class R = MakeVectorImpl::vector_return_type<T0, Ts...>>
+R make_vector(Ts&&... ts) {
```

Comment:
Much of this file seems to be devoted to extracting the first type in `Ts`.  Since this overload is only used for one or more arguments, the first type could be a separate template argument and then I think most of the above reduces to a std::conditional_t.